pulsar-common/src/test/java/org/apache/pulsar/common/nar/NarUnpackerTest.java

```diff
@@ -118,6 +118,28 @@ public static void main(String[] args) {
         }
     }
 
+    @Test
+    void shouldReExtractWhenUnpackedDirectoryIsMissing() throws InterruptedException {
+        CountDownLatch countDownLatch = new CountDownLatch(1);
+        AtomicInteger exceptionCounter = new AtomicInteger();
+        AtomicInteger extractCounter = new AtomicInteger();
+
+        try {
+            File narWorkingDirectory = NarUnpacker.doUnpackNar(sampleZipFile, extractDirectory, extractCounter::incrementAndGet);
+            FileUtils.deleteFile(narWorkingDirectory, true);
+            NarUnpacker.doUnpackNar(sampleZipFile, extractDirectory, extractCounter::incrementAndGet);
+        } catch (Exception e) {
+            log.error("Unpacking failed", e);
+            exceptionCounter.incrementAndGet();
+        } finally {
+            countDownLatch.countDown();
+        }
```

Comment:
remove this. there's no need to catch the exception since this isn't running in a background thread. if an exception is thrown, the test will fail.